optuna/_hypervolume/hssp.py

```diff
@@ -5,6 +5,40 @@
 import optuna
 
 
+def _solve_hssp_2d(
+    rank_i_loss_vals: np.ndarray,
+    rank_i_indices: np.ndarray,
+    subset_size: int,
+    reference_point: np.ndarray,
+) -> np.ndarray:
+    # This function can be used for non-unique rank_i_loss_vals as well.
+    # The time complexity is O(subset_size * rank_i_loss_vals.shape[0]).
+    assert rank_i_loss_vals.shape[-1] == 2 and subset_size <= rank_i_loss_vals.shape[0]
+    n_trials = rank_i_loss_vals.shape[0]
+    order = np.argsort(rank_i_loss_vals[:, 0])
+    sorted_loss_vals = rank_i_loss_vals[order]
+    # The diagonal points for each rectangular to calculate the hypervolume contributions.
+    rect_diags = np.repeat(reference_point[np.newaxis, :], n_trials, axis=0)
+    subset_indices = np.zeros(subset_size, dtype=int)
+    for i in range(subset_size):
+        contribs = np.prod(rect_diags - sorted_loss_vals, axis=-1)
+        max_index = np.argmax(contribs)
+        subset_indices[i] = rank_i_indices[order[max_index]]
+        loss_vals = sorted_loss_vals[max_index].copy()
+
+        keep = np.ones(n_trials - i, dtype=bool)
+        keep[max_index] = False
+        # Remove the chosen point.
+        order = order[keep]
+        rect_diags = rect_diags[keep]
+        sorted_loss_vals = sorted_loss_vals[keep]
+        # Update the diagonal points for each hypervolume contribution calculation.
+        rect_diags[:max_index, 0] = np.minimum(loss_vals[0], rect_diags[:max_index, 0])
+        rect_diags[max_index:, 1] = np.minimum(loss_vals[1], rect_diags[max_index:, 1])
+
+    return subset_indices
```

Comment:
How about aligning it with the implementation for N-dimension?
```suggestion
    selected_indices = np.zeros(subset_size, dtype=int)
    for i in range(subset_size):
        contribs = np.prod(rect_diags - sorted_loss_vals, axis=-1)
        max_index = np.argmax(contribs)
        selected_indices[i] = order[max_index]
        loss_vals = sorted_loss_vals[max_index].copy()

        keep = np.ones(n_trials - i, dtype=bool)
        keep[max_index] = False
        # Remove the chosen point.
        order = order[keep]
        rect_diags = rect_diags[keep]
        sorted_loss_vals = sorted_loss_vals[keep]
        # Update the diagonal points for each hypervolume contribution calculation.
        rect_diags[:max_index, 0] = np.minimum(loss_vals[0], rect_diags[:max_index, 0])
        rect_diags[max_index:, 1] = np.minimum(loss_vals[1], rect_diags[max_index:, 1])

    return rank_i_indices[selected_indices]
```